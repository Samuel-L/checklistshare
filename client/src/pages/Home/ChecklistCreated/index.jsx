import React from 'react';
import PropTypes from 'prop-types';

import Grid from 'material-ui/Grid';
import Typography from 'material-ui/Typography';
import { CircularProgress } from 'material-ui/Progress';

import clientBaseURL from '../../../utils/client-helpers';

const ChecklistCreated = props => (
  <Grid container justify="center">
    { props.adding && !props.added
      ?
        <Grid item>
          <CircularProgress size={70} />
        </Grid>
      :
        <Grid item>
          <Typography
            variant="headline"
            align="center"
            gutterBottom
          >
            Checklist created!
          </Typography>
          <Typography
            variant="body2"
            align="center"
          >
            {clientBaseURL}/{props.url}
          </Typography>
        </Grid>
    }
  </Grid>
);

ChecklistCreated.propTypes = {
  adding: PropTypes.bool,
  added: PropTypes.bool,
  url: PropTypes.string,
};

ChecklistCreated.defaultProps = {
  adding: false,
  added: false,
  url: '',
};

export default ChecklistCreated;
