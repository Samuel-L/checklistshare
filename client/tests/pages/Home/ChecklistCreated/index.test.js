import React from 'react';
import renderer from 'react-test-renderer';

import ChecklistCreated from '../../../../src/pages/Home/ChecklistCreated';

describe('pages/Home: ChecklistCreated', () => {
  const tree = renderer.create(<ChecklistCreated />).toJSON();

  it('matches snapshot', () => {
     expect(tree).toMatchSnapshot();
  });
});
