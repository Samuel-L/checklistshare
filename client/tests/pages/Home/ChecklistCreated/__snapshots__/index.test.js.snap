// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Home: ChecklistCreated matches snapshot 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-justify-xs-center-18"
>
  <div
    className="MuiGrid-item-2"
  >
    <h1
      className="MuiTypography-root-91 MuiTypography-headline-96 MuiTypography-gutterBottom-108 MuiTypography-alignCenter-104"
    >
      Checklist created!
    </h1>
    <aside
      className="MuiTypography-root-91 MuiTypography-body2-99 MuiTypography-alignCenter-104"
    >
      https://www.changemewhenurlishere.com
      /
      
    </aside>
  </div>
</div>
`;
